OpenRA.Mods.Common/FileFormats/WavReader.cs

```diff
@@ -206,15 +208,19 @@ public sealed class WavStreamMsAdpcm : ReadOnlyAdapterStream
 
 			static readonly int[] AdaptCoeff2 = [0, -256, 0, 64, 0, -208, -232];
 
+			readonly int dataSize;
 			readonly short channels;
 			readonly int blockDataSize;
 			readonly int numBlocks;
 
 			int currentBlock;
 
+			public override long Length => dataSize;
+
 			public WavStreamMsAdpcm(Stream stream, int dataSize, short blockAlign, short channels)
 				: base(stream)
 			{
+				this.dataSize = dataSize;
 				this.channels = channels;
 				blockDataSize = blockAlign - channels * 7;
 				numBlocks = dataSize / blockAlign;
```

Comment:
If we're after some memory improvements, we should declare the various arrays (`bpred`, `chanIdelta`, `s1`, `s2`) at the class level and reuse them.